File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        const int n =1e5 + 10 ;
        unordered_map<int , bool> mp;
        for (int i = 0; i < nums.size(); i++) {
            if (mp[nums[i]])
                return true;
            mp[nums[i]] = true;
        }
        return false ;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-3.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ret = []
        nums.sort()
        for i in range(1 , len(nums) - 1):
            l , r  = i - 1 , i + 1
            while l >= 0 and r < len(nums):
                if nums[i] + nums[l] + nums[r] == 0:
                    if not ret or [nums[l] , nums[i] , nums[r]] not in ret :
                        ret.append([nums[l] , nums[i] , nums[r]])
                    r += 1
                elif nums[i] + nums[l] + nums[r] > 0 :
                    l -= 1
                else :
                    r += 1  
        return ret
        